inference-engine/src/mkldnn_plugin/mkldnn_node.cpp

```diff
@@ -1537,6 +1551,69 @@ bool MKLDNNNode::canBePerformedAsScaleShift(const MKLDNNNode *parentNode) const
             || isConvertablePowerStatic();
 }
 
+bool MKLDNNNode::inputShapesDefined() const {
+    for (size_t i = 0; i < getParentEdges().size(); i++) {
+        if (!getParentEdgesAtPort(i)[0]->getMemory().getDesc().isDefined())
+            return false;
+    }
+    return true;
+}
+
+bool MKLDNNNode::needPrepareParams() const {
+    return inputShapesModified();
+}
+
+bool MKLDNNNode::inputShapesModified() const {
+    if (lastInputDims.size() != getParentEdges().size()) {
+        if (lastInputDims.empty())
+            return true;
+        IE_THROW() << "Input dims and parent edges number mismatch!";
+    }
```

Comment:
```suggestion
    if (lastInputDims.empty())
        return true;
    if (lastInputDims.size() != getParentEdges().size()) {
        IE_THROW() << "Input dims and parent edges number mismatch!";
    }
```

Reply:
changes not valid for node without input edges